mlflow/models/evaluation/default_evaluator.py

```diff
@@ -971,25 +916,86 @@ def plot_confusion_matrix():
+            self.y_pred = self.predict_fn(self.X.copy_to_avoid_mutation())
+            self.is_binomial = self.num_classes <= 2
 
-    def _evaluate_regressor(self):
-        self.y_pred = self.model.predict(self.X.copy_to_avoid_mutation())
-        self.metrics.update(_get_regressor_metrics(self.y, self.y_pred))
+            if self.is_binomial:
+                if list(self.label_list) not in [[0, 1], [-1, 1]]:
+                    raise ValueError(
+                        "Binary classifier evaluation dataset positive class label must be 1 or"
+                        " True, negative class label must be 0 or -1 or False, and dataset"
+                        " must contains both positive and negative examples."
+                    )
+                _logger.info(
+                    "The evaluation dataset is inferred as binary dataset, positive label is "
+                    f"{self.label_list[1]}, negative label is {self.label_list[0]}."
+                )
+            else:
+                _logger.info(
+                    "The evaluation dataset is inferred as multiclass dataset, number of classes "
+                    f"is inferred as {self.num_classes}"
+                )
+
+            if self.predict_proba_fn is not None:
+                self.y_probs = self.predict_proba_fn(self.X.copy_to_avoid_mutation())
+                if self.is_binomial:
+                    self.y_prob = self.y_probs[:, 1]
+                else:
+                    self.y_prob = None
+            else:
+                self.y_probs = None
+                self.y_prob = None
+        elif self.model_type == "regressor":
+            self.y_pred = self.model.predict(self.X.copy_to_avoid_mutation())
+
+    def _compute_builtin_metrics(self):
+        """
+        Helper method for computing builtin metrics for self.model, update results to self.metrics
+        """
         self._evaluate_sklearn_model_score_if_scorable()
+        if self.model_type == "classifier":
+            self.metrics.update(
+                _get_classifier_global_metrics(
+                    self.is_binomial,
+                    self.y,
+                    self.y_pred,
+                    self.y_probs,
+                    self.label_list,
+                )
+            )
+            if self.is_binomial:
+                self.metrics.update(_get_classifier_per_class_metrics(self.y, self.y_pred))
+                self._compute_roc_and_pr_curve()
+        elif self.model_type == "regressor":
+            self.metrics.update(_get_regressor_metrics(self.y, self.y_pred))
 
-        return self._log_and_return_evaluation_result()
+    def _log_metrics_and_artifacts(self):
+        """
+        Helper method for generating artifacts, logging metrics and artifacts.
+        """
+        if self.model_type == "classifier":
+            if self.is_binomial:
+                self._log_binary_classifier()
+            else:
+                self._log_multiclass_classifier()
```

Comment:
Should these two function names be updated to be explicitly about logging artifacts? Otherwise, I find the functions a bit confusing.